Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -817,8 +817,7 @@ def pow (k : ℕ) : ArithmeticFunction ℕ :=
 theorem pow_apply {k n : ℕ} : pow k n = if k = 0 ∧ n = 0 then 0 else n ^ k := by
   cases k
   · simp [pow]
-  rename_i k  -- Porting note: added
-  simp [pow, k.succ_pos.ne']
+  rfl
```

Comment:
I guess this is fine, but please double-check.

Reply:
I suppose this works because `n.succ = 0` gets reduced away?

An alternate proof for the whole theorem: `by cases k <;> simp [pow]`.